doc/compiler/optimizer/OptimizerDesignFeatures.md

```diff
@@ -0,0 +1,303 @@
+the basic aliasing that is there relies more on properties of the fundamental
+properties of the symbols in use, e.g. a call aliases all static symbols.
+
+Of course, the aliasing in particular can be extended or improved fairly easily and the optimizations in
+the compiler are shielded from the internals of how the aliasing is computed
+via well-defined APIs.
+
+Interprocedural analysis is another concept that is common in static compilers but once again the payoffs
+in a dynamic language for a JIT compiler are probably not worth the effort to build and maintain a call
+graph and associated data structures.
+
+Another cool capability in the OpenJ9 JIT compiler that somewhat softens the blow of having to run
+optimizations iteratively is the capability that optimizations must enable other optimizations as they run.
+In other words, an optimization can either create or detect an opportunity that makes it important to run
+another optimization, but the infrastructure allows later optimization passes to be enabled on-demand rather
+having to be run unconditionally, meaning we only pay the compile time of repeating optimizations if there
+is some expected benefit.
+
+## IL Representation
+
+The OpenJ9 JIT compiler uses a (Directed Acyclic Graph) DAG to represent individual computations in
+the program. A JIT compiler probably can only afford to run many expensive optimizations locally (i.e., per
+extended basic block) given the compile time constraints it operates under, and the DAG representation is
+particularly convenient for doing local optimizations.
+
+The DAG representation is less expensive from a memory footprint viewpoint since a single value within an
+extended basic block is represented by a single node regardless of how many consumers (parent nodes in the
+OpenJ9 JIT IL representation) of the value there are in that extended basic block.
+
+It also makes the concept of "value equality" trivial for most local analyses, since this basically just
+boils down to "node equality" making the code much cleaner and obviates the need for a separate analysis
+step that tracks "value equality".
+
+Another aspect of the OpenJ9 JIT IL representation that is novel is the use of exception check opcodes
+and exception edges in the Control Flow Graph (CFG).
+
+Together these two notions avoid chopping up the compiled method into many small basic blocks ending every
+time an exception check must be done.
+
+Essentially exception check opcodes represent control flow exiting the extended basic block going to the
+exception catch block (if there exists one in the method) and exception edges represent this flow in the CFG.
+
+Crucially this is done without ending the basic block every time an exception check is encountered, and the
+"special" nature of exception checks and edges is known to the compiler's generic
+[dataflow analysis](https://youtu.be/YCCdJ1Qphao) framework, so that it does the right operations every time
```

Comment:
Would a link to the OMR [Data-Flow Engine](https://github.com/eclipse/omr/blob/master/doc/compiler/optimizer/DataFlowEngine.md) be appropriate here?